packages/utils/src/calculate-gas-fees.ts

```diff
@@ -0,0 +1,50 @@
+import { BigNumber, constants, providers } from 'ethers';
+import { suggestFees } from 'eip1559-fee-suggestions-ethers';
+import Utils from '.';
+
+/**
+ * The function calculates gas fee information relative to EIP-1559.
+ * @param provider Generic provider.
+ * @param fallbackToGasPrice A boolean allowing if true to return gasPrice in case maxFeePerGas and maxPriorityFeePerGas equals 0.
+ * @param gasPriceMin Minimum gas price to return.
+ * @returns an object containing:
+ * - maxFeePerGas: The maximum fee per unit of gas for this transaction.
+ *   maxFeePerGas = baseFeePerGas + maxPriorityFeePerGas
+ *   The baseFeePerGas depends on how full the previous blocks were.
+ * - maxPriorityFeePerGas: The maximum priority fee per unit of gas for this transaction.
+ * - gasPrice: Optional fallback: the gas price for this transaction.
+ */
+async function calculateGasFees(
+  provider: providers.JsonRpcProvider,
+  fallbackToGasPrice = false,
+  gasPriceMin?: BigNumber,
```

Comment:
I didn't get the meaning of `true` or `false` until I saw the method. Could we change the signature to an object instead? It would make it clearer:
```suggestion
  {
    provider,
    fallbackToGasPrice = false,
    gasPriceMin,
  } : {
    provider: providers.JsonRpcProvider,
    fallbackToGasPrice: boolean,
    gasPriceMin?: BigNumber,
  }
```